profiler/src/main/java/com/splunk/opentelemetry/profiler/EventProcessingChain.java

```diff
@@ -58,12 +58,12 @@ void accept(RecordedEvent event) {
     eventStats.incEventCount();
     switch (eventName) {
       case ContextAttached.EVENT_NAME:
-        if (chunkTracker.contextAttached(event)) {
-          flushBuffer();
+        try (EventTimer eventTimer = eventStats.time(eventName)) {
+          spanContextualizer.updateContext(event);
         }
-        buffer.add(event);
         break;
       case ThreadDumpProcessor.EVENT_NAME:
+        // TODO: is that correct?
         if (chunkTracker.threadDump(event)) {
```

Comment:
@breedx-splk @laurit I noticed that chunking was completely breaking span linkage in TLAB events (after I added it there); we should probably do an all-or-nothing approach here and either include TLABs or remove chunking altogether.

Reply:
this got resolved by changing the chunking logic